tools/buildsys/src/gomod/error.rs

```diff
@@ -0,0 +1,34 @@
+use snafu::Snafu;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(super)))]
+pub(crate) enum Error {
+    #[snafu(display("Failed to start command: {}", source))]
+    CommandStart { source: std::io::Error },
+
+    #[snafu(display("Failed to execute command: 'docker {}'", args))]
+    DockerExecution { args: String },
+
+    #[snafu(display("input is required"))]
+    InputFile,
+
+    #[snafu(display("input must be a file and end with .tar.gz"))]
+    InputFileBad,
```

Comment:
nit: we don't seem to check for the "ends with .tar.gz" part (and might not want to, since other compression formats might be used)